src/GraphQL/Types/Composite/AutoRegisteringHelper.cs

```diff
@@ -303,5 +303,65 @@ internal static TypeInformation GetTypeInformation(ParameterInfo parameterInfo)
             typeInformation.ApplyAttributes();
             return typeInformation;
         }
+
+        /// <summary>
+        /// Identifies the constructor to use when constructing instances of <typeparamref name="TSourceType"/>.
+        /// Selects any public constructor marked with <see cref="GraphQLConstructorAttribute"/>, or the public
+        /// parameterless constructor, or the only public contructor, or returns <see langword="null"/> otherwise.
+        /// </summary>
+        internal static ConstructorInfo? GetConstructorOrDefault<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TSourceType>()
+            => GetConstructorOrDefault(typeof(TSourceType));
+
+        /// <summary>
+        /// Identifies the constructor to use when constructing instances of <paramref name="sourceType"/>.
+        /// Selects any public constructor marked with <see cref="GraphQLConstructorAttribute"/>, or the public
+        /// parameterless constructor, or the only public contructor, or returns <see langword="null"/> otherwise.
+        /// </summary>
+        internal static ConstructorInfo? GetConstructorOrDefault([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type sourceType)
+        {
+            var constructors = sourceType.GetConstructors();
+            // if there are no public constructors, return null
+            if (constructors == null || constructors.Length == 0)
+                return null;
+            // if there is only one public constructor, return it
+            if (constructors.Length == 1)
+                return constructors[0];
+            // if there are multiple public constructors, return the one marked with GraphQLConstructorAttribute, or the parameterless constructor, or null
+            ConstructorInfo? match = null;
+            ConstructorInfo? parameterless = null;
+            foreach (var constructor in constructors)
+            {
+                if (constructor.GetCustomAttribute<GraphQLConstructorAttribute>() != null)
+                {
+                    if (match != null)
+                        throw new InvalidOperationException($"Multiple constructors marked with {nameof(GraphQLConstructorAttribute)} found on type '{sourceType.GetFriendlyName()}'.");
```

Comment:
This is always a design error and should not occur, regardless of whether `ParseDictionary` is overridden.